source/common/upstream/load_balancer_impl.cc

```diff
@@ -1299,19 +1299,80 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
                                                                 const HostsSource&) {
   HostSharedPtr candidate_host = nullptr;
 
+  switch (selection_method_) {
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest::FULL_SCAN:
+      candidate_host = unweightedHostPickFullScan(hosts_to_use);
+      break;
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest::N_CHOICES:
+      candidate_host = unweightedHostPickNChoices(hosts_to_use);
+      break;
+    default:
+      IS_ENVOY_BUG("unknown selection method specified for least request load balancer");
+  }
+
+  return candidate_host;
+}
+
+HostSharedPtr LeastRequestLoadBalancer::unweightedHostPickFullScan(const HostVector& hosts_to_use) {
+  HostSharedPtr candidate_host = nullptr;
+
+  uint32_t num_hosts_known_tied_for_least = 0;
+
+  const uint32_t num_hosts = hosts_to_use.size();
```

Comment:
nit: Use `size_t` instead of `uint32_t`.

Reply:
In that case, I'll replace all the `uint32_t`s with `size_t`s in this function